var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -109,7 +109,7 @@ class Cp2k(MakefilePackage, CudaPackage, CMakePackage, ROCmPackage):
         " are enabled",
     )
     variant("pytorch", default=False, description="Enable libtorch support")
```

Comment:
As install with pytorch only implemented with cmake, I suggest add a "when=build_system=cmake" here.

Reply:
+shared is enforced for both build systems. that the best solution